src/connection.ts

```diff
@@ -3718,54 +3713,46 @@ Connection.prototype.STATE = {
 
           if (fedAuthInfoToken && fedAuthInfoToken.stsurl && fedAuthInfoToken.spn) {
             const authentication = this.config.authentication as AzureActiveDirectoryPasswordAuthentication | AzureActiveDirectoryMsiVmAuthentication | AzureActiveDirectoryMsiAppServiceAuthentication | AzureActiveDirectoryServicePrincipalSecret;
+            const tokenScope = fedAuthInfoToken.spn;
 
             const getToken = (callback: (error: Error | null, token?: string) => void) => {
-              const getTokenFromCredentials = (err: Error | undefined, credentials?: UserTokenCredentials | MSIAppServiceTokenCredentials | MSIVmTokenCredentials | ApplicationTokenCredentials) => {
-                if (err) {
-                  return callback(err);
-                }
-
-                credentials!.getToken().then((tokenResponse: { accessToken: string | undefined }) => {
-                  callback(null, tokenResponse.accessToken);
-                }, callback);
-              };
+              const getTokenFromCredentials = (credentials: TokenCredential) =>
+                credentials.getToken([ tokenScope ]).then(
+                  (tokenResponse) => callback(null, tokenResponse?.token),
+                  callback
+                );
```

Comment:
Would you mind adding braces here for readability?

```suggestion
              const getTokenFromCredentials = (credentials: TokenCredential) => {
                credentials.getToken([ tokenScope ]).then(
                  (tokenResponse) => callback(null, tokenResponse?.token),
                  callback
                );
              };
```